### DISCUSSION TITLE
Need help using a Sphinx instance for GitHub Page

### DISCUSSION THREAD

**Discussion by jaygray0919** [community/community#52982] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I have a Sphinx repository https://github.com/jaygray0919/ewb

The contents are a successful local build, found here: https://github.com/jaygray0919/ewb/blob/main/_build/html/index.html

Following the GitHub Page instructions, I copied those files here: https://github.com/jaygray0919/ewb/tree/main/docs

In GitHub Settings/Pages I see this statement:
`Your GitHub Pages site is currently being built from the /docs folder in the main branch.`

But the page build generates a 404

How would I point to either:
https://github.com/jaygray0919/ewb/blob/main/_build/html/index.html
`or`
https://github.com/jaygray0919/ewb/blob/main/docs/index.html

My preference is to point here: https://github.com/jaygray0919/ewb/blob/main/_build/html/index.html
Reason: that directory is generated by the local build and is automatically updated after a local build.

**Comment by airtower-luna** [comment, marked as answer]
> In GitHub Settings/Pages I see this statement:
> Your GitHub Pages site is currently being built from the /docs folder in the main branch.

That's probably misleading, because you're using your own Actions workflow to deploy, not the default one. The workflow [is configured to deploy the whole repository](https://github.com/jaygray0919/ewb/blob/6f906f957333a6647c928141342a6a47c11333c8/.github/workflows/static.yml#L36-L39). That https://jaygray0919.github.io/ewb/docs/ looks like a Sphinx site confirms it.

You could change the `path` parameter for `actions/upload-pages-artifact` to `docs/`, or (more elegant) instead of committing the built files run the Sphinx build in the workflow and then point the upload action at the result. :slightly_smiling_face:

**Reply by jaygray0919** [reply]
OK, I'll try: `run the Sphinx build in the workflow and then point the upload action at the result` I see these:
  1.  https://github.com/sphinx-doc/sphinx/issues/3382#issuecomment-1059362139
       with `publish_dir: docs/_build/html` (my configuration) `and`
  2.  https://www.sphinx-doc.org/en/master/tutorial/deploying.html `and`
  3. https://github.com/uibcdf/action-sphinx-docs-to-gh-pages `and`
  4. https://github.com/marketplace/actions/sphinx-to-github-pages

Not sure if there is a difference with 
  *  uses: actions/checkout@v2
and
  *  uses: actions/checkout@v3

Thanks for the guidance.

**Reply by airtower-luna** [reply]
> Not sure if there is a difference with
> 
>     * uses: actions/checkout@v2
>       and
> 
>     * uses: actions/checkout@v3

This is the simple part of the answer: `v2` is an outdated version of `actions/checkout`, use `v3` (the part behind the `@` is a Git ref or commit ID). :wink: 

Of the links you posted, 1 through 3 seem to use the old method of pushing the build to a branch.

4 looks very interesting, because it seems to wrap all the steps you need into one action (the "Publishing with this action" section in the docs). Looks worth a try. :slightly_smiling_face: 

The other option is to write the build steps yourself: Use [`actions/setup-python`](https://github.com/actions/setup-python), install Sphinx, run the Sphinx build (both `run` steps), and point `actions/upload-pages-artifact` at the output directory.
